src/include/storage/block_layout.h

```diff
@@ -25,7 +26,7 @@ class BlockLayout {
    *
    * @param attr_sizes vector of attribute sizes.
    */
-  explicit BlockLayout(std::vector<uint8_t> attr_sizes);
+  explicit BlockLayout(std::vector<uint16_t> attr_sizes);
```

Comment:
Are we likely to ever need to change this? Should we make a `#define` for it so that we know all the locations where it is referenced in the code?

Reply:
@mbutrovich probably has a better idea than me?

Reply:
Do we anticipate inline attributes larger than 32KB? Moving to uint32_t for attr_sizes would necessitate a change to block size.

Reply:
We require that each block can hold at least one tuple. Up until now we have been statically defining a "max number of columns" such that even if all columns are the largest possible, a block can still hold at least one tuple.

This one probably already breaks the old value, but we have asserts in place to catch it when it goes wrong. I actually believe that checking on the fly is preferable to statically fixing a max number of columns. The code to do so is already in there as asserts and tests. 

I can file an issue for this if we believe that is the way forward.